doc/proxy-metrics.md

```diff
@@ -133,3 +135,63 @@ request_total{
 
 [prom-format]: https://prometheus.io/docs/instrumenting/exposition_formats/#format-version-0.0.4
 [pod-template-hash]: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#pod-template-hash-label
+
+# Transport-Level Metrics
+
+The following metrics are collected at the level of the underlying transport
+layer.
+
+### `connection_opens`
+
+A counter of the total number of transport connections which have been opened.
+
+### `connection_closes`
+
+A counter of the total number of transport connections which have been closed.
+
+### `connections`
+
+A gauge of the number of transport connections currently open.
+
+### `sent_bytes`
+
+A counter of the total number of sent bytes.
+
+### `received_bytes`
+
+A counter of the total number of recieved bytes.
+
+### `connection_duration_ms`
+
+A histogram of the duration of the lifetime of a connection, in milliseconds.
+
+## Labels
+
+Each of these metrics has the following labels:
+
+* `direction`: `inbound` if the connection was established either from outside the
+                pod to the proxy, or from the proxy to the application,
+               `outbound` if the connection was established either from the
+                application to the proxy, or from the proxy to outside the pod.
+* `kind`: `client` if the connection was initiated by the proxy,
+          `server` if the connection was initiated to the proxy.
+* `protocol`: `http` if the connection corresponds to an HTTP/1 or HTTP/2 request,
+              `tcp` if the connection is proxied as a raw TCP stream.
+
+Note that the labels described above under the heading "Prometheus Collector labels"
+are also added to transport-level metrics, when applicable.
+
+
+### Outbound HTTP labels
+
+The following labels are only applicable if `direction="outbound"`, `protocol="http",`
+_and_ `kind="server"`.
+
+* `dst_deployment`: The deployment to which this request is being sent.
+* `dst_job`: The job to which this request is being sent.
```

Comment:
this should be `dst_k8s_job`.

more generally, these labels are identical to the protocol-level outbound labels. would we expect them to always be the same? noting that `kind="server"` does not exist in the protocol-level labels, which set of data (`kind="server"` vs. `kind="client"`) would correspond to protocol-level labels?

Reply:
@siggy yes, that's correct. These are the protocol-level outbound labels. The concept of `kind={"server", "client"}` doesn't apply to protocol-level metrics, as a request or response encompasses both the connections to and from the proxy. 

However, we can only associate the destination labels with server connections, as when the proxy accepts an incoming client connection, it has yet to determine the target of the protocol-level request (if there is one).